stubs/PyYAML/yaml/error.pyi

```diff
@@ -1,21 +1,28 @@
-from typing import Any
+from typing import Optional
 
 class Mark:
-    name: Any
-    index: Any
-    line: Any
-    column: Any
-    buffer: Any
-    pointer: Any
-    def __init__(self, name, index, line, column, buffer, pointer) -> None: ...
-    def get_snippet(self, indent=..., max_length=...): ...
+    name: str
+    index: int
+    line: int
+    column: int
+    buffer: Optional[str]
+    pointer: int
+    def __init__(self, name: str, index: int, line: int, column: int, buffer: Optional[str], pointer: int) -> None: ...
+    def get_snippet(self, indent: int = ..., max_length: int = ...) -> Optional[str]: ...
 
 class YAMLError(Exception): ...
 
 class MarkedYAMLError(YAMLError):
-    context: Any
-    context_mark: Any
-    problem: Any
-    problem_mark: Any
-    note: Any
-    def __init__(self, context=..., context_mark=..., problem=..., problem_mark=..., note=...) -> None: ...
+    context: Optional[str]
+    context_mark: Optional[Mark]
+    problem: Optional[str]
+    problem_mark: Optional[Mark]
+    note: Optional[str]
```

Comment:
When using an instance of this class, do we expect users to check whether each attribute is `None`? With this code, they have to.

Reply:
@Akuli not sure what you mean - the types are consistent with the current implementation.

Reply:
You can't `print("note: " + error.note)`, for example. Mypy would complain that this can become `print("note: " + None)`, which is an error. Is this intentional?

Reply:
@Akuli I know what the optional type means - shouldn't I derive the signature from the usage [in the actual impl](https://github.com/yaml/pyyaml/blob/ee37f4653c08fc07aecff69cfd92848e6b1a540e/lib3/yaml/scanner.py#L491-L493) rather than constructing my own one to satisfy `print("spam " + error.context)`?

Reply:
Union return types can be annoying, as explained in `CONTRIBUTING.md`. Attributes are similar to return values in this sense.

After thinking about this a bit more, I think your annotations are good. If you catch the exception and do `print("spam " + error.context)`, it really is a problem: the `None` can occur in practice, and it's bad if you don't handle it.

Reply:
@Akuli I understand - in my own stubs and custom `YAMLError` impls, I don't require `None` types because I don't validate the stubs against the `PyYAML` implementation. My concern is that if I don't do that in `typeshed`, it fails the stubtest. Is that ok? If yes, I'll gladly remove the optionals in `Mark` and `MarkedYAMLError`.

Reply:
Currently the best way to say "this can be string or None, but it's fine to assume it's string" is `str | Any`. It looks a bit weird, but it does the job, and it's not same as just `Any`: `foo + 123` fails if `foo` has type `str | Any`, but succeeds if it has type `Any`. This also won't fail stubtest.

I don't have a strong opinion on this. I don't think I have ever used `MarkedYAMLError`. If you think it is useful to just use the attributes without checking for `None`-ness, you can make another PR that changes them to `str | Any`.